test/hack/bin/verify-rules.sh

```diff
@@ -0,0 +1,200 @@
+
+            # Syntax check of rules file
+            echo "###    promtool check rules $GIT_WORKDIR/test/tests/providers/$provider/$filename"
+            local promtool_check_output
+            if ! promtool_check_output="$("$GIT_WORKDIR/$PROMTOOL" check rules "$GIT_WORKDIR/test/tests/providers/$provider/$filename" 2>&1)";
+            then
+                echo "###   Syntax check failing for $file:"
+                echo "$promtool_check_output"
+                promtool_check_errors+=("$promtool_check_output")
+                continue
+            fi
+            checked_files+=("$file")
+
+
+            local global_testfile="$GIT_WORKDIR/test/tests/providers/global/${filename%.yml}.test.yml"
+            local provider_testfile="$GIT_WORKDIR/test/tests/providers/$provider/${filename%.yml}.test.yml"
+
+
+            # if the file is whitelisted via the global ignore file
+            if array_contains "$file" "${expected_failure_prefixes_global[@]}"; then
+                # don't run tests for it
+                echo "###  Skipping $file: listed in $expected_failure_relative_file_global"
+                continue
+            fi
+            # if the file is whitelisted via the provider ignore file
+            if array_contains "$file" "${expected_failure_prefixes_provider[@]}"; then
+                # don't run tests for it
+                echo "###  Skipping $file: listed in $expected_failure_relative_file_provider"
+                continue
+            fi
+
+            # don't run tests if no provider specific tests are defined
+            if [[ ! -d "$GIT_WORKDIR/test/tests/providers/$provider" ]]; then
+                echo "###   No tests for proviter $provider - skipping"
+                continue
+            fi
+
+            # Fail if no testfile found
+            if [[ ! -f "$global_testfile" ]] \
+                && [[ ! -f "$provider_testfile" ]]
+            then
+                echo "### No testfile found for $filename - error"
+                promtool_test_errors+=("NO TEST FILE for $filename")
+                continue
+            fi
+
+            local promtool_test_output
+
+            if [[ -f "$global_testfile" ]]; then
+                echo "###    promtool test rules ${filename%.yml}.test.yml - global"
+                promtool_test_output="$("$GIT_WORKDIR/$PROMTOOL" test rules "$global_testfile" 2>&1)" ||
+                    promtool_test_errors+=("$promtool_test_output")
+            fi
+
+            if [[ -f "$provider_testfile" ]]; then
+                echo "###    promtool test rules ${filename%.yml}.test.yml - $provider"
+                promtool_test_output="$("$GIT_WORKDIR/$PROMTOOL" test rules "$provider_testfile" 2>&1)" ||
+                    promtool_test_errors+=("$promtool_test_output")
+            fi
+
+        done
+    done
+
+    # Cleanup
+    git checkout -- "$GIT_WORKDIR/helm/prometheus-rules/Chart.yaml"
+    git checkout -- "$GIT_WORKDIR/helm/prometheus-rules/values.yaml"
```

Comment:
I am not really keen on changing the current git state here. I would prefer that all changes happen in a different folder.

Reply:
we have to run `architect helm template` to get a valid helm template which is required to generate templated alerting rules.

Reply:
I would prefer that we copy the chart folder somewhere in a tmp directory and run `architect helm template` there. Then we can safely trash the test folder.

Reply:
just learned that `architect` has the `--working-directory` flag ... should be a easy change, will do that.

Reply:
We tried the `--working-directory` flag, which is not working as we expected. It yield a `not a git repository` error. So we decided to keep changing things in the current directory and move the destructive commands under `make clean`.